syncer/syncer_test.go

```diff
@@ -222,33 +243,117 @@ func (s *testSyncerSuite) TestSelectTable(c *C) {
 			{Schema: "~^ptest*", Name: "~^t.*"},
 		},
 	}
+	s.resetEventsGenerator()
```

Comment:
We can define a more generic object to represent the test case, like
```
type binlogEvents []binlogEvent
type binlogEvent struct {
  typ string
  arguments []interface{}
}
```
and an events generation function
```
function testGenerateEvent(binlogEvents) []*replication.BinlogEvent
```
they would save many codes

Reply:
Done